azerothcore/azerothcore-wotlk
# fix(Core/Scripts): Rewrite Hungarfen

## Changes Proposed:
-  Remove unnecessary things from the instance cpp.
-  Move bosses from ScriptedAI to BossAI
-  Move Hungarfen from SAI to cpp.
-  Move Underbog mushrooms from SAI to cpp.
-  Make Hungarfen uninterruptable.
-  Fixed stupid 0.0001 value set on speed_walk for both mushrooms

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14398
- Closes https://github.com/chromiecraft/chromiecraft/issues/4687
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/13979
- Closes https://github.com/chromiecraft/chromiecraft/issues/4662

## SOURCE:

Sniffs

## Tests Performed:

- Tested in-game.

## How to Test the Changes:


1. Engage Hungarfen and see if everything is alright.
2. Check Ghazan and The Black Stalker just in case.

## Known Issues and TODO List:


- [x] Needs to be merged after mod scale PR.
